src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -296,7 +315,48 @@ private void loadPathToMainScript() {
         this.appiumJS = findMainScript();
     }
 
+    private String parseCapabilitiesIfWindows() {
+        String result = StringUtils.EMPTY;
+        final List<String> pathCapabilities = ImmutableList.of(AndroidMobileCapabilityType.KEYSTORE_PATH,
+                AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE, MobileCapabilityType.APP);
+
+        if (capabilities != null) {
+            Map<String, ?> capabilitiesMap = capabilities.asMap();
+            Set<? extends Map.Entry<String, ?>> entries = capabilitiesMap.entrySet();
+
+            for (Map.Entry<String, ?> entry : entries) {
+                Object value = entry.getValue();
+
+                if (value == null) {
+                    continue;
+                }
+
+                if (String.class.isAssignableFrom(value.getClass())) {
```

Comment:
value instanceof String should be enough

Reply:
resolved in the newest push